### PR TITLE
Fixes [#24664] Improve System Message Formatting for Channel Header Edits

### PR DESCRIPTION
**Pull request by ZubairImtiaz3**
#### Summary
This PR addresses the issue where system messages, when editing a channel header, were displayed on a single line. The change ensures that multiple paragraphs are correctly displayed on separate lines, consistent with the channel header's appearance.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/24664
Jira https://mattermost.atlassian.net/browse/MM-53997

```release-note
Improved readability by displaying system messages on multiple lines after editing a channel header. #24664
```

#### Screenshots
Before: 
<a href="https://ibb.co/4mcthsS">Screenshot-2023-10-05-124937</a>

2)
<a href="https://ibb.co/KhGDdq8">Screenshot-2023-10-05-133204</a>

After: 
<a href="https://ibb.co/zxHPQyz">Screenshot-2023-10-05-125038</a>

### CODE REVIEW

**Review by harshilsharma63 — APPROVED** [review]
Looks good!

File: webapp/channels/src/sass/components/_post.scss [hunk 1]
```diff
@@ -1475,7 +1475,7 @@
         }
 
         p + p {
-            margin: 0.5em 0 0;
+            margin: 0 0 0;
```
Comment (larkox) [inline webapp/channels/src/sass/components/_post.scss:1478]: Won't this change change how every posts will look like? Is this something desired?

File: webapp/channels/src/sass/layout/_markdown.scss [hunk 2]
```diff
@@ -41,8 +41,6 @@ h6.markdown__heading {
 }
 
 .markdown__paragraph-inline {
-    display: inline;
```
Comment (larkox) [inline webapp/channels/src/sass/layout/_markdown.scss:44]: Why is this needed? How will this change how other things are displayed as of today?

File: webapp/channels/src/components/post_markdown/system_message_helpers.tsx [hunk 3]
```diff
@@ -195,8 +195,16 @@ function renderHeaderChangeMessage(post: Post): ReactNode {
...
     const username = renderUsername(post.props.username);
-    const oldHeader = post.props.old_header ? renderFormattedText(post.props.old_header, headerOptions, post) : null;
-    const newHeader = post.props.new_header ? renderFormattedText(post.props.new_header, headerOptions, post) : null;
+    const oldHeader = post.props.old_header ? renderFormattedText(
+        post.props.old_header.replace(/\n/g, '\n\n'),
+        headerOptions,
+        post,
+    ) : null;
+    const newHeader = post.props.new_header ? renderFormattedText(
+        post.props.new_header.replace(/\n/g, '\n\n'),
```
Comment (larkox) [inline webapp/channels/src/components/post_markdown/system_message_helpers.tsx:204]: Why is this needed? wouldn't removing the `singleline` from the options on line 192 solve the issue?

**Review by larkox — CHANGES_REQUESTED** [review]
A few questions

**Comment by ZubairImtiaz3** [comment]
@larkox Hi, thank you for your insights. Initially, I made the CSS adjustments based on certain formatting nuances I observed. However, after your feedback, I revisited the issue and realized that simply removing the single line from line 192 resolves the matter without the need for the CSS changes. Your advice was invaluable in refining this PR.

File: webapp/channels/src/components/post_markdown/system_message_helpers.tsx (hunk 3 above)
Comment (ZubairImtiaz3) [reply webapp/channels/src/components/post_markdown/system_message_helpers.tsx]: Yes, you're right. Simply removing the single line from line 192 resolves the matter.

**Review by harshilsharma63 — APPROVED** [review]

**Review by larkox — APPROVED** [review]
Great! Thanks!

**Review by abhijit-singh — APPROVED** [review]
Thanks @ZubairImtiaz3! Looks good!

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by larkox** [comment]
Thank you for your contribution!
